fast64_internal/oot/data/oot_object_data.py

```diff
@@ -0,0 +1,89 @@
+from dataclasses import dataclass
+from os import path
+from .oot_getters import getXMLRoot, getEnumList
+from .oot_data import OoT_BaseElement
+
+# Note: "object" in this context refers to an OoT Object file (like ``gameplay_keep``)
+
+
+@dataclass
+class OoT_ObjectElement(OoT_BaseElement):
+    pass
+
+
+class OoT_ObjectData:
+    """Everything related to OoT objects"""
+
+    def __init__(self):
+        # general object list
+        self.objectList: list[OoT_ObjectElement] = []
+
+        # list of tuples used by Blender's enum properties
+        self.ootEnumObjectKey: list[tuple] = []
+        self.ootEnumObjectIDLegacy: list[tuple] = []  # for old blends
+
+        # Path to the ``ObjectList.xml`` file
+        objectXML = path.dirname(path.abspath(__file__)) + "/xml/ObjectList.xml"
+
+        for obj in getXMLRoot(objectXML).iterfind("Object"):
+            objName = f"{obj.attrib['Name']} - {obj.attrib['ID'].replace('OBJECT_', '')}"
+            self.objectList.append(OoT_ObjectElement(obj.attrib["ID"], obj.attrib["Key"], objName))
+
+        self.objectsByID = {obj.id: obj for obj in self.objectList}
+        self.objectsByKey = {obj.key: obj for obj in self.objectList}
+        self.ootEnumObjectKey, self.ootEnumObjectIDLegacy = getEnumList(self.objectList, "Custom Object")
+
+    def upgradeObjectList(self, objList):
+        for obj in objList:
+            obj.objectKey = self.objectsByID.get(obj.objectID).key
+
+    def upgradeAltHeaders(self, roomObj):
+        altHeaders = roomObj.ootAlternateRoomHeaders
+        for header in ["childNightHeader", "adultDayHeader", "adultNightHeader"]:
+            curHeader = getattr(altHeaders, header)
+            if curHeader is not None:
+                self.upgradeObjectList(curHeader.objectList)
+        for i in range(len(altHeaders.cutsceneHeaders)):
+            self.upgradeObjectList(altHeaders.cutsceneHeaders[i].objectList)
+
+    def addMissingObjectToUI(self, roomObj, headerIndex, objectKey, csHeaderIndex):
```

Comment:
I don't like that name, props are also stored in the blend to be saved, not just being displayed
no suggestion for now though